### PR TITLE
add a description field for GatewayClass

### PR DESCRIPTION
**Pull request by hzliangbin**
**What type of PR is this?**

**What this PR does / why we need it**:
add a field to help describe gatewayclass  

**Which issue(s) this PR fixes**:

Fixes #
#610 

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: apis/v1alpha2/gatewayclass_types.go [hunk 1]
```diff
@@ -117,6 +118,13 @@ type ParametersReference struct {
 	// +kubebuilder:validation:MaxLength=253
 	// +optional
 	Namespace *string `json:"namespace,omitempty"`
+
+	// Description helps describe a GatewayClass with more details.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +optional
+	Description *string `json:"description,omitempty"`
```
Comment (hbagdi) [inline apis/v1alpha2/gatewayclass_types.go:127]: I think this field should be on the spec struct.

**Review by tao12345666333 — COMMENTED** [review]
/lgtm

**Review by hbagdi — COMMENTED** [review]
/cc @robscott @bowei @jpeach

File: apis/v1alpha2/gatewayclass_types.go [hunk 2]
```diff
@@ -79,6 +80,13 @@ type GatewayClassSpec struct {
 	//
 	// +optional
 	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
+
+	// Description helps describe a GatewayClass with more details.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
```
[the remark this answers is not in the document]
Comment (robscott) [reply apis/v1alpha2/gatewayclass_types.go:87]: Although I'd be open to expanding this in the future if needed, I think a primary goal of this was to enable helpful kubectl output, so something that could fit in a single row would be ideal.

File: apis/v1alpha2/gatewayclass_types.go [hunk 3]
```diff
@@ -28,6 +28,7 @@ import (
 // +kubebuilder:storageversion
 // +kubebuilder:printcolumn:name="Controller",type=string,JSONPath=`.spec.controller`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:printcolumn:name="Description",type=string,JSONPath=`.spec.description`
```
Comment (robscott) [inline apis/v1alpha2/gatewayclass_types.go:31]: We may want to set a priority field here. That would enable us to [only include this in the wide output](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#priority).

```suggestion
// +kubebuilder:printcolumn:name="Description",type=string,JSONPath=`.spec.description`,priority=1
```

File: apis/v1alpha2/gatewayclass_types.go [hunk 4]
```diff
@@ -79,6 +80,13 @@ type GatewayClassSpec struct {
 	//
 	// +optional
 	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
+
+	// Description helps describe a GatewayClass with more details.
+	//
+	// +kubebuilder:validation:MinLength=1
```
Comment (robscott) [inline apis/v1alpha2/gatewayclass_types.go:86]: It feels a bit confusing to enforce a min length on an optional field.

**Review by robscott — COMMENTED** [review]
Thanks for your work on this @hzliangbin!

File: apis/v1alpha2/gatewayclass_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bowei) [reply apis/v1alpha2/gatewayclass_types.go:87]: 128 isn't that much more -- we aren't using VT100's anymore...

**Comment by hbagdi** [comment]
/lgtm cancel
@hzliangbin Can you please re-run `./hack/update-codegen.sh`?

**Pull request merged**